Addition of Tamil parser

anshsharma10 opened the pull request:
Tamil parser can be run from the command line to parse Tamil corpora and split them into sentences.

anshsharma10 committed:
Create tamil parser

Tamil parser can be run from the command line to parse Tamil corpora and split them into sentences.

Natkeeran reviewed (commented):
Thanks @pivot5 

Can we split this into smaller functions.  It will help with readability and help us extend and maintain as well.  

pseudocode follows:

stats = {}

sentences:List[str] = get_common_voice_sentences(file_path: str)

file_content:str = get_file_content(file_path: str)
sentences::List[str] = get_sentences(text: str)
stats['total_sentences_before_processing'] = total_sentences_before_processing

sentences::List[str] = process_senences(senences:List[str])
stats['total_sentences_after_processing'] = total_sentences_after_processing

result:boolean = check_english_characters(sentence:str)
result:boolean = check_numbers_in_dateformat(sentence:str)
sentence:str = remove_symbols(sentence:str)
sentence:str = replace_number_with_numeral(sentence:str)
result:boolean = check_words_length(sentence:str, min_words_length:int, max_words_length:int)

anshsharma10 committed:
Update tamil_parser.py

anshsharma10 committed:
Make efficiency changes & add functions

anshsharma10 committed:
Efficiency improvement

Conversion of processing to single loop to improve efficiency, and addition of functions to improve readability

Natkeeran commented:
@pivot5 I going to merge this code and refactor it little bit.  Thank you.
